Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+                new_domain = Domain(
+                    unique_by_name_preserve_order(flatten(t.domain.attributes for t in tables)),
+                    unique_by_name_preserve_order(flatten(t.domain.class_vars for t in tables)),
+                    unique_by_name_preserve_order(flatten(t.domain.metas for t in tables))
+                )
+            new_index = cls._new_id(len(result))
+            result.index = new_index
+        elif axis == CONCAT_COLS:
+            # check for same name
+            columns = [v for v in flatten([list(t.columns) for t in tables]) if v not in cls._INTERNAL_COLUMN_NAMES]
+            if len(set(columns)) != len(columns):
+                raise ValueError("Cannot concatenate domains with same names.")
+            if colstack:
+                # check for same length
+                if len(set(len(t) for t in tables)) != 1:
+                    raise ValueError("Cannot colstack tables with differing numbers of rows. ")
+                # reset index temporarily because this joins by index by default
+                result = pd.concat([t.reset_index(drop=True) for t in tables], axis=1, join_axes=[tables[0].index])
+            else:
+                result = pd.concat(tables, axis=1)
+
+            # merge domains
+            new_domain = Domain(
+                list(flatten(t.domain.attributes for t in tables)),
+                list(flatten(t.domain.class_vars for t in tables)),
+                list(flatten(t.domain.metas for t in tables))
+            )
+
+            # fix multiple weight columns: keep the first one
+            weight_columns = result[cls._WEIGHTS_COLUMN]
+            result = result.drop(cls._WEIGHTS_COLUMN, axis=1)
+            result[cls._WEIGHTS_COLUMN] = weight_columns.iloc[:, 0]
+
+            if reindex:
+                new_index = cls._new_id(len(result))
+                result.index = new_index
+        else:
+            raise ValueError('axis {} out of bounds [0, 2)'.format(axis))
+        result._transfer_properties(tables[0])  # pd.concat does not do this by itself
+        return cls.from_dataframe(new_domain, result)
+
+    def _transfer_properties(self, from_table, transfer_domain=False):
+        """Transfer properties (such as the name) to this table.
+
+        Parameters
+        ----------
+        from_table : TableBase
+            The table to source the attributes from.
+        transfer_domain : bool, optional, default False
+            Whether to transfer the domain.
+
+        Notes
+        -----
+        This should normally not be used, but it is used when these properties
+        are not automatically transferred on manipulation, in particular when using pd.concat.
+        """
+        for name in self._metadata:
+            if hasattr(from_table, name) and (transfer_domain or name != "domain"):
+                setattr(self, name, getattr(from_table, name))
+
+    def approx_len(self):
```

Comment:
Hooray for Spark!

But I don't know of anyone who reads wikis. This is TableBase. Add a `TODO: remove` marker,  possibly with that link, into the code.